plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -878,23 +878,26 @@ func haveSame(a1, a2 []string) bool {
 
 func GeneralPredicates(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
 	var predicateFails []algorithm.PredicateFailureReason
-	fit, reasons, err := noncriticalPredicates(pod, meta, nodeInfo)
+	fit, reasons, err := EssentialPredicates(pod, meta, nodeInfo)
 	if err != nil {
 		return false, predicateFails, err
 	}
 	if !fit {
+		// No need check other predicates if the essential predicates failed
 		predicateFails = append(predicateFails, reasons...)
+		return len(predicateFails) == 0, predicateFails, nil
```

Comment:
@wojtek-t The reason that I was trying to return early is that I was trying to ignore the predicate failing caused by `PodFitsResources`, if we get all predicate failures, then the end user will always get the error returned by `PodFitsResources` even if the user already set `nodeSelector`, this is the issue reported by #45780 , any comments for how to avoid such case? Thanks.